pkg/reconciler/source/crd/crd_test.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package crd
+
+import (
+	"context"
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"knative.dev/eventing/pkg/apis/sources"
+	"knative.dev/pkg/client/injection/ducks/duck/v1/source"
+	"knative.dev/pkg/configmap"
+	"knative.dev/pkg/controller"
+	logtesting "knative.dev/pkg/logging/testing"
+
+	. "knative.dev/eventing/pkg/reconciler/testing"
+	. "knative.dev/pkg/reconciler/testing"
+
+	// Fake injection informers
+	_ "knative.dev/eventing/pkg/client/injection/informers/eventing/v1alpha1/eventtype/fake"
+	_ "knative.dev/pkg/client/injection/apiextensions/informers/apiextensions/v1beta1/customresourcedefinition/fake"
+	_ "knative.dev/pkg/client/injection/ducks/duck/v1/source/fake"
+)
+
+const (
+	crdName          = "test-crd"
+	crdGroup         = "testing.sources.knative.dev"
+	crdKind          = "TestSource"
+	crdPlural        = "testsources"
+	crdVersionServed = "v1alpha1"
+)
+
+func TestAllCases(t *testing.T) {
+	table := TableTest{{
+		Name: "bad workqueue key",
+		// Make sure Reconcile handles bad keys.
+		Key: "too/many/parts",
+	}, {
+		Name: "key not found",
+		// Make sure Reconcile handles good keys that don't exist.
+		Key: "not-found",
+	}, {
+		Name: "reconcile failed, cannot find GVR or GVK",
+		Objects: []runtime.Object{
+			NewCustomResourceDefinition(crdName,
+				WithCustomResourceDefinitionLabels(map[string]string{
+					sources.SourceDuckLabelKey: sources.SourceDuckLabelValue,
+				})),
+		},
+		Key:     crdName,
+		WantErr: true,
+		WantEvents: []string{
+			Eventf(corev1.EventTypeWarning, "SourceCRDReconcileFailed", "Source CRD reconciliation failed: unable to find GVR or GVK for %s", crdName),
+		},
+	},
+	//}, {
+	// TODO uncomment the following once we figure out why the eventtype informer is missing from the duck.controller.
```

Comment:
@n3wscott I'm having trouble with this one. 
It seems that the context we pass to the Source duck controller doesn't have eventtype: https://github.com/knative/eventing/blob/7dca1dfa529eaf7009ccbd9db34171dcf45b013a/pkg/reconciler/source/crd/crd.go#L203, although I'm importing the fake...
I might be missing something, OR there is a bug in https://github.com/knative/eventing/blob/7dca1dfa529eaf7009ccbd9db34171dcf45b013a/pkg/reconciler/source/crd/crd.go#L201... 
Any ideas?

Reply:
What is the error that is thrown?

Reply:
does it not have the kind EventType? I need more info to try to help

Reply:
sorrry, was AFK...
The problem that the informer is not registered: `Unable to fetch knative.dev/eventing/pkg/client/informers/externalversions/eventing/v1alpha1.EventTypeInformer from context`

updated the codegen stuff... will be back in a few mins